sandy/sections/mf6.py

```diff
@@ -0,0 +1,515 @@
+     1001.00000 9.986200-1          0          6          1          2 128 6 16    6
+              2          2                                             128 6 16    7
+     3339002.00 1.00000000  150000000 1.00000000                       128 6 16    8
+     2.99862000 0.00000000          0          0          0          3 128 6 16    9
+
+    LAW 7:
+    >>> tape = sandy.get_endf6_file("endfb_71", 'xs', 40090)
+    >>> sec = read_mf6(tape, 425, 16)
+    >>> text = write_mf6(sec)
+    >>> print(text[:1000])
+     4009.00000 8.93478000          0          1          2          0 425 6 16    1
+     1.00000000 1.00000000          0          7          1         17 425 6 16    2
+             17          1                                             425 6 16    3
+     1748830.00 2.00000000 2700000.00 2.00000000 3250000.00 2.00000000 425 6 16    4
+     3900000.00 2.00000000 4500000.00 2.00000000 5900000.00 2.00000000 425 6 16    5
+     6400000.00 2.00000000 7050000.00 2.00000000 8400000.00 2.00000000 425 6 16    6
+     10100000.0 2.00000000 11000000.0 2.00000000 12000000.0 2.00000000 425 6 16    7
+     13000000.0 2.00000000 14200000.0 2.00000000 15400000.0 2.00000000 425 6 16    8
+     17000000.0 2.00000000 20000000.0 2.00000000                       425 6 16    9
+     0.00000000 0.00000000          0          0          1         18 425 6 16   10
+             18          2                                             425 6 16   11
+     0.00000000 1748830.00          0          0          1         21 425 6 16   12
+             21          2
+
+    """
+    lines = sandy.write_cont(
+                sec["ZA"],
+                sec["AWR"],
+                0,
+                sec['LCT'],
+                len(sec["NK"]),
+                0,
+                )
+    for product, NK in sec["NK"].items():  # For the rest of the NK subsections
+        # [MAT, 6, MT/ ZAP, AWP, LIP, LAW,NR,NP/Eint/yi(E)]TAB1
+        ZAP, LIP = zam.zam2za(product)
+        lines += sandy.write_tab1(
+                    ZAP,
+                    NK["AWP"],
+                    LIP,
+                    NK["LAW"],
+                    NK["NR"],
+                    NK["NP"],
+                    NK["E"],
+                    NK["Y"],
+                    )
+        # LAW dependent structures:
+        if NK["LAW"] == 1:
+            # [MAT, 6, MT/ 0.0, 0.0, LANG, LEP, NR, NE/ Eint]TAB2 for each subsection
+            lines += sandy.write_tab2(
+                            0,
+                            0,
+                            NK["LANG"],
+                            NK["LEP"],
+                            NK["ENR"],
+                            NK["ENE"],
+                            NK["EINT"],
+                    )
+            # repeat the LIST structure for all the NE incident energies
+            for hx in range(NK["ENE"][0]):
+                    key = list(NK["EGROUPS"].keys())
```

Comment:
don't use this, if you loop on a dictionary use `.items()`, like for MF10

Reply:
Sorry, i upload the old version insted off the new one with the diccionary iterations